### ISSUE TITLE
Token getting disabled frequently

### ISSUE THREAD

**Issue by dipamchang** [bitpay/java-bitpay-client#5]
Hello, 
I am integrating Bitpay as a payment gateway for our company to accept in Bitcoin. 
But the issue I am facing (in development) is, Whenever I generate a Invoive, the token gets disabled, and the next time I make any request to Bitpay server , I get regular "Info: Pair this client with your merchant account using the pairing code:****". 

Can you please guide me as to what I am doing wrong ?

**Comment by philosodad** [comment]
Can you give me a little more information? For example, how do you know that the token is being disabled? Also, how are you saving/loading your private key between sessions, and how are you storing your token?

**Comment by dipamchang** [comment]
Ok, so what I was doing is making new BitPay object for each new session. And hence it was unable to authenticate previously generated token. Now I am initializing the Bitpay in a static block, so that its shared among all the sessions. 
So is this the right way to do ?

**Comment by philosodad** [comment]
That will work. In other libraries, the typical approach is to save the key and token and recreate the client using those as inputs to the constructor, which is an alternate approach that might be appropriate for some use cases.

**Comment by dipamchang** [comment]
Thanks a lot for the answer. I am a newbie in this field, and this was definitely a new information gained. I'll modify the code to implement this method, as it looks more clean.
Thanks.

**Comment by dipamchang** [comment]
Just one more thing, during the implementation I observed that the Bitpay server is returning a field key called "flags" in the response for new Invoice generation. This field is not present in the Bean to which the Jackson is mapping the JSON and hence breaks throwing an exception. I fixed it in my implementation of this library. So just wanted to ask if you accept Pull Requests so that we can fix this, before anyone faces the same issue.
